### PR TITLE
RavenDB-17262 - Creation of setup ZIP file from RVN implemented. 

### PR DESCRIPTION
**Pull request by omerlewitz**
### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-17262

### Additional description
It has been tested locally from `RVN` command and from the `setup wizard` and it worked. 
Regarding the ticket - `-u|--unsecured-ravendb-url http://localhost:3456  # optional` why do we need that?

**I noticed I ruined some indentations I will push a change tomorrow.**

### CODE REVIEW

File: src/Raven.Server/Commercial/LetsEncryptUtils.cs [hunk 1]
```diff
@@ -0,0 +1,1159 @@
...
+        public static async Task<byte[]> SetupLetsEncryptByRvn(SetupInfo setupInfo, string settingsPath , CancellationToken token)
+        {
+            Console.WriteLine("Setting up RavenDB in Let's Encrypt security mode.");
+
+            if (SetupManager.IsValidEmail(setupInfo.Email) == false)
+                throw new ArgumentException("Invalid e-mail format" + setupInfo.Email);
+
+            var acmeClient = new LetsEncryptClient(AcmeClientUrl);
+
+            Console.WriteLine($"Getting challenge(s) from Let's Encrypt. Using e-mail: {setupInfo.Email}.");
```
Comment (ayende) [inline src/Raven.Server/Commercial/LetsEncryptUtils.cs:55]: Nope

File: src/Raven.Server/Commercial/LetsEncryptUtils.cs [hunk 2]
```diff
@@ -0,0 +1,1159 @@
...
+            await acmeClient.Init(setupInfo.Email, token);
+
+            var challengeResult = await InitialLetsEncryptChallenge(setupInfo, acmeClient, token);
+            Console.WriteLine(challengeResult.Challenge != null
+                ? "Successfully received challenge(s) information from Let's Encrypt."
+                : "Using cached Let's Encrypt certificate.");
+
+            Console.WriteLine($"Updating DNS record(s) and challenge(s) in {setupInfo.Domain.ToLower()}.{setupInfo.RootDomain.ToLower()}.");
```
Comment (ayende) [inline src/Raven.Server/Commercial/LetsEncryptUtils.cs:63]: Nope, etc.

File: src/Raven.Server/ServerWide/SecretProtection.cs [hunk 3]
```diff
@@ -325,8 +326,26 @@ private static byte[] DecryptProtectedData(byte[] secret, byte[] key)
...
 
 #if !RVN
+        public static  LetsEncryptUtils.CertificateHolder ValidateCertificateAndCreateCertificateHolder(string source, X509Certificate2 loadedCertificate, byte[] rawBytes, string password, LicenseType licenseType, bool certificateValidationKeyUsages)
+        {
+            ValidateExpiration(source, loadedCertificate,licenseType);
+
+            ValidatePrivateKey(source, password, rawBytes, out var privateKey);
+
+            ValidateKeyUsages(source, loadedCertificate, certificateValidationKeyUsages);
+            //ValidateKeyUsages(source, loadedCertificate, serverStore.Configuration.Security.CertificateValidationKeyUsages);
```
Comment (ayende) [inline src/Raven.Server/ServerWide/SecretProtection.cs:336]: Why is this commented?

**Review by ayende — CHANGES_REQUESTED** [review]

**Comment by omerlewitz** [comment]
@ayende  it's a mess.
I can't Rebase and there's a lot of indentation problems.
Working on a new PR and I will attach the link later on.

**Comment by omerlewitz** [comment]
New PR here -https://github.com/ravendb/ravendb/pull/13408

**Pull request closed without merging**